token-validation-core/src/main/java/no/nav/security/token/support/core/validation/CasualJwtTokenValidator.java

```diff
@@ -0,0 +1,74 @@
+package no.nav.security.token.support.core.validation;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.JWSVerificationKeySelector;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jose.util.ResourceRetriever;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import com.nimbusds.jwt.proc.DefaultJWTProcessor;
+import com.nimbusds.jwt.proc.JWTClaimsSetVerifier;
+import no.nav.security.token.support.core.exceptions.JwtTokenValidatorException;
+
+import java.net.URL;
+import java.util.HashSet;
+import java.util.List;
+
+public class CasualJwtTokenValidator implements JwtTokenValidator {
```

Comment:
Not sold on the name "casual" (naming is hard). I was thinking we might go for a "configurable" validator with sane defaults (as we briefly discussed in the mob prog session) - as such the class could be named `ConfigurableJwtTokenValidator` or something like that - and take a list of required claims from config (other than iss, nbf and exp).

Reply:
Agree, as always, naming is hard. But seems like the better choice.